addon-resizer/healthcheck/healthcheck.go

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package healthcheck
+
+import (
+	"fmt"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/golang/glog"
+)
+
+// HealthCheck contains information about last activity time of the monitored component.
+// NOTE: This started as a simplified version of VPA's HealthCheck.
+type HealthCheck struct {
+	// configuration
+	address         string
+	activityTimeout time.Duration
+	// current state
+	lastActivity time.Time
+	mutex        *sync.Mutex
```

Comment:
nit: I would switch mutex with lastActivity.
In k8s there is often added a mutex and all variables guarded by it are below it.

Reply:
Switched (here & below).